### DISCUSSION TITLE
[kf5*] update to kf6* (Whether to change the name of the ports)

### DISCUSSION THREAD

**Discussion by xahon** [microsoft/vcpkg#42555] [Q&A]
### Library name

kf5*

### New version number

kf6*

### Other information that may be useful (release notes, etc...)

kf5archive
kf5attica
kf5auth
kf5bookmarks
kf5codecs
kf5completion
kf5config
kf5configwidgets
kf5coreaddons
kf5crash
kf5dbusaddons
kf5declarative
kf5diagram
kf5globalaccel
kf5guiaddons
kf5holidays
kf5i18n
kf5iconthemes
kf5itemmodels
kf5itemviews
kf5jobwidgets
kf5kcmutils
kf5kio
kf5newstuff
kf5notifications
kf5package
kf5parts
kf5plotting
kf5service
kf5solid
kf5sonnet
kf5syntaxhighlighting
kf5texteditor
kf5textwidgets
kf5wallet
kf5widgetsaddons
kf5windowsystem
kf5xmlgui

to

kf6archive
kf6attica
kf6auth
kf6bookmarks
kf6codecs
kf6completion
kf6config
kf6configwidgets
kf6coreaddons
kf6crash
kf6dbusaddons
kf6declarative
kf6diagram
kf6globalaccel
kf6guiaddons
kf6holidays
kf6i18n
kf6iconthemes
kf6itemmodels
kf6itemviews
kf6jobwidgets
kf6kcmutils
kf6kio
kf6newstuff
kf6notifications
kf6package
kf6parts
kf6plotting
kf6service
kf6solid
kf6sonnet
kf6syntaxhighlighting
kf6texteditor
kf6textwidgets
kf6wallet
kf6widgetsaddons
kf6windowsystem
kf6xmlgui

**Comment by equeim** [comment]
FYI you may encounter these issues with static triplets on Windows:

https://invent.kde.org/frameworks/extra-cmake-modules/-/merge_requests/360

https://bugs.kde.org/show_bug.cgi?id=481954

**Comment by MonicaLiu0311** [comment]
The upstream library name is:
```
KDE/karchive
KDE/attica
KDE/kauth
...
```

The name in vcpkg is:
```
kf5archive
kf5attica
kf5auth
...
```

When `kf5*` was added to vcpkg as a [new port](https://github.com/microsoft/vcpkg/pull/4302/files#diff-988b6ce968d91525ee03247045da9382305244dd7eb6aac16916867f367d6da3), the upstream was in the process of releasing `5.x.x`, so the user added the character `"f5"` to the port name. As a result, when the upstream releases to `6.x.x`, new users propose to update to `kf6*`, but this will change the names of `38` ports, and there will be other versions in the future... Therefore, convert this issue into a discussion to explore how to handle it better.

**Comment by felixf4xu** [comment]
my opinion: keep both kf5* and kf6* ports.

If you worry about kf7*, that is reasonable concern, but we need to accept it if qt5/qt6 and qt7 exists at the same time.
